tests/test_typechecker.py

```diff
@@ -2651,8 +2651,14 @@ class TestTypecheck_module(unittest.TestCase):
     def test_function_py2(self):
         from testhelpers import modulewide_typecheck_testhelper_py2 as mth
         self.assertEqual(mth.testfunc(3, 2.5, 'abcd'), (9, 7.5))
+        with self.assertRaises(KeyError):
+            pytypes.typechecked_module('nonexistent123')
         self.assertEqual(mth.testfunc(3, 2.5, 7), (9, 7.5)) # would normally fail
-        pytypes.typechecked_module(mth)
+        module_name = 'testhelpers.modulewide_typecheck_testhelper_py2'
+        returned_mth = pytypes.typechecked_module(module_name)
+        self.assertEqual(returned_mth, module_name)
+        returned_mth = pytypes.typechecked_module(mth)
+        self.assertEqual(returned_mth, mth)
```

Comment:
I added test coverage but had to jam all the logic into one test case to make it pass. If you want it split up into smaller individual test cases, please LMK how you'd suggest to structure it. I wrote it as smaller individual test cases at first but couldn't get it to pass since once typechecking is enabled on the module, the global state of whether it's enabled leaks into other tests using the same module. Could define more testhelper modules one per test, or have some kind of teardown to deactivate type checking, etc., but what approach would be a good cost-to-benefit ratio?

Reply:
Current solution is fine. It's actually more than I expected. Thanks!